src/apps/Altinn.AccessManagement/src/Altinn.AccessMgmt.Core/HostedServices/Services/AllAltinnRoleSyncService.cs

```diff
@@ -0,0 +1,209 @@
+                        {
+                            _logger.LogWarning("Ingest partial complete: Assignment ({0}/{1})", ingested, batchData.Count);
+                        }
+
+                        var merged = await ingestService.MergeTempData<Assignment>(batchId, previousOptions, GetAssignmentMergeMatchFilter, cancellationToken: cancellationToken);
+
+                        _logger.LogInformation("Merge complete: Assignment ({0}/{1})", merged, ingested);
+                    }
+                    catch (Exception ex)
+                    {
+                        _logger.LogError(ex, "Failed to ingest and/or merge Assignment and EntityLookup batch {0} to db", batchName);
+                        throw new Exception(string.Format("Failed to ingest and/or merge Assignment and EntityLookup batch {0} to db", batchName), ex);
+                    }
+                    finally
+                    {
+                        batchId = Guid.CreateVersion7();
+                        batchData.Clear();
+                    }
+                }
+            }
+        }
+
+        private static readonly IReadOnlyList<string> GetAssignmentMergeMatchFilter = new List<string>() { "fromid", "roleid", "toid" }.AsReadOnly();
+
+        private async Task<(Assignment Assignment, AuditValues Options)> ConvertRoleDelegationModelToAssignment(AppDbContext dbContext, RoleDelegationModel model, string batchId, CancellationToken cancellationToken)
+        {
+            try
+            {
+                var role = await GetRole(dbContext, model.RoleTypeCode, cancellationToken);
+
+                Assignment assignment = new Assignment()
+                {
+                    Id = Guid.CreateVersion7(),
+                    FromId = model.FromPartyUuid,
+                    ToId = model.ToUserPartyUuid.Value,
+                    RoleId = role.Id,
+                    Audit_ValidFrom = model.DelegationChangeDateTime?.UtcDateTime ?? DateTime.UtcNow
+                };
+                
+                AuditValues options = new AuditValues(model.PerformedByPartyUuid ?? model.PerformedByUserUuid ?? SystemEntityConstants.Altinn2RoleImportSystem, SystemEntityConstants.Altinn2RoleImportSystem, batchId, model.DelegationChangeDateTime ?? DateTimeOffset.UtcNow);
+
+                return (assignment, options);
+            }
+            catch (Exception ex)
+            {
+                throw new Exception(string.Format("Failed to convert model to Assignment. From:{0} To:{1} Role:{2}", model.FromPartyUuid, model.ToUserPartyUuid, model.RoleTypeCode));
+            }
+        }
+
+        private Dictionary<string, Role> Roles { get; set; } = new Dictionary<string, Role>(StringComparer.OrdinalIgnoreCase);
+
+        private async Task<Role> GetRole(AppDbContext dbContext, string roleCode, CancellationToken cancellationToken)
+        {
+            string roleIdentifier = $"urn:altinn:rolecode:{roleCode}";
```

Comment:
er roleCode fra A2 her uppercase? for vi har vel ellers normalisert oss på lowercase